Please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/108488/artifacts?artifactName=failure%20logs%20for%20x64-windows&api-version=7.1&%24format=zip) on `x64-widnows`:
```
[265/266] C:\Windows\system32\cmd.exe /C "C:\Windows\system32\cmd.exe /C "D:\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E __create_def D:\b\7zip\x64-windows-dbg\CMakeFiles\7zip.dir\.\exports.def D:\b\7zip\x64-windows-dbg\CMakeFiles\7zip.dir\.\exports.def.objs && cd D:\b\7zip\x64-windows-dbg" && D:\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=CMakeFiles\7zip.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe  @CMakeFiles\7zip.rsp  /out:7zip.dll /implib:7zip.lib /pdb:7zip.pdb /dll /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  /DEF:CMakeFiles\7zip.dir\.\exports.def && cd ."
FAILED: 7zip.dll 7zip.lib 
C:\Windows\system32\cmd.exe /C "C:\Windows\system32\cmd.exe /C "D:\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E __create_def D:\b\7zip\x64-windows-dbg\CMakeFiles\7zip.dir\.\exports.def D:\b\7zip\x64-windows-dbg\CMakeFiles\7zip.dir\.\exports.def.objs && cd D:\b\7zip\x64-windows-dbg" && D:\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=CMakeFiles\7zip.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe  @CMakeFiles\7zip.rsp  /out:7zip.dll /implib:7zip.lib /pdb:7zip.pdb /dll /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  /DEF:CMakeFiles\7zip.dir\.\exports.def && cd ."
LINK Pass 1: command "C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe @CMakeFiles\7zip.rsp /out:7zip.dll /implib:7zip.lib /pdb:7zip.pdb /dll /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL /DEF:CMakeFiles\7zip.dir\.\exports.def /MANIFEST /MANIFESTFILE:CMakeFiles\7zip.dir/intermediate.manifest CMakeFiles\7zip.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library 7zip.lib and object 7zip.exp
InOutTempBuffer.cpp.obj : error LNK2019: unresolved external symbol "long __cdecl GetLastError_noZero_HRESULT(void)" (?GetLastError_noZero_HRESULT@@YAJXZ) referenced in function "public: long __cdecl CInOutTempBuffer::Write_HRESULT(void const *,unsigned int)" (?Write_HRESULT@CInOutTempBuffer@@QEAAJPEBXI@Z)
InOutTempBuffer.cpp.obj : error LNK2019: unresolved external symbol "public: bool __cdecl NWindows::NFile::NIO::CFileBase::Close(void)" (?Close@CFileBase@NIO@NFile@NWindows@@QEAA_NXZ) referenced in function "public: long __cdecl CInOutTempBuffer::WriteToStream(struct ISequentialOutStream *)" (?WriteToStream@CInOutTempBuffer@@QEAAJPEAUISequentialOutStream@@@Z)
InOutTempBuffer.cpp.obj : error LNK2019: unresolved external symbol "public: bool __cdecl NWindows::NFile::NIO::CInFile::Open(wchar_t const *)" (?Open@CInFile@NIO@NFile@NWindows@@QEAA_NPEB_W@Z) referenced in function "public: long __cdecl CInOutTempBuffer::WriteToStream(struct ISequentialOutStream *)" (?WriteToStream@CInOutTempBuffer@@QEAAJPEAUISequentialOutStream@@@Z)
```

--- a/ports/7zip/CMakeLists.txt
+++ b/ports/7zip/CMakeLists.txt
@@ -149,16 +149,6 @@ target_sources(7zip PRIVATE
     CPP/Common/XzCrc64Reg.cpp
     CPP/Common/Xxh64Reg.cpp
 
-    CPP/Windows/FileDir.cpp
-    CPP/Windows/FileFind.cpp
-    CPP/Windows/FileIO.cpp
-    CPP/Windows/FileName.cpp
-    CPP/Windows/PropVariant.cpp
-    CPP/Windows/PropVariantUtils.cpp
-    CPP/Windows/Synchronization.cpp
-    CPP/Windows/System.cpp
-    CPP/Windows/TimeUtils.cpp
-
     CPP/7zip/Common/CreateCoder.cpp
     CPP/7zip/Common/CWrappers.cpp
     CPP/7zip/Common/InBuffer.cpp
@@ -301,6 +291,20 @@ target_sources(7zip PRIVATE
     CPP/7zip/Archive/Archive2.def
     C/Util/LzmaLib/LzmaLib.def
 )
+
+if(WIN32)
+    target_sources(7zip PRIVATE
+        CPP/Windows/FileDir.cpp
+        CPP/Windows/FileFind.cpp
+        CPP/Windows/FileIO.cpp
+        CPP/Windows/FileName.cpp
+        CPP/Windows/PropVariant.cpp
+        CPP/Windows/PropVariantUtils.cpp
+        CPP/Windows/Synchronization.cpp
+        CPP/Windows/System.cpp
+        CPP/Windows/TimeUtils.cpp
+    )
+endif()
 
 # 7zCrcOpt
 if(VCPKG_TARGET_IS_WINDOWS)

--- a/ports/7zip/vcpkg.json
+++ b/ports/7zip/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "7zip",
   "version-string": "24.08",
+  "port-version": 1,
   "description": "Library for archiving file with a high compression ratio.",
   "homepage": "https://www.7-zip.org",
   "license": "LGPL-2.1-or-later",

--- a/versions/7-/7zip.json
+++ b/versions/7-/7zip.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "36bee800b8aadc1908cc02cc1561dee40a3b43f6",
+      "version-string": "24.08",
+      "port-version": 1
+    },
+    {
       "git-tree": "f8021d6c213215f0eb9e99500d348291ac5898c5",
       "version-string": "24.08",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6,7 +6,7 @@
     },
     "7zip": {
       "baseline": "24.08",
-      "port-version": 0
+      "port-version": 1
     },
     "ableton": {
       "baseline": "3.0.6",

